graph/appender/util.go

```diff
@@ -20,16 +21,18 @@ func checkError(err error) {
 	}
 }
 
-func promQuery(query string, queryTime time.Time, api v1.API) model.Vector {
+func promQuery(query string, queryTime time.Time, api v1.API, a Appender) model.Vector {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
 	// wrap with a round() to be in line with metrics api
 	query = fmt.Sprintf("round(%s,0.001)", query)
 	log.Debugf("Appender query:\n%s&time=%v (now=%v, %v)\n", query, queryTime.Format(graph.TF), time.Now().Format(graph.TF), queryTime.Unix())
 
+	promtimer := internalmetrics.GetPrometheusProcessingTimePrometheusTimer("Graph-Appender-" + a.Name())
 	value, err := api.Query(ctx, query, queryTime)
 	checkError(err)
+	promtimer.ObserveDuration() // notice we only collect metrics for successful prom queries
```

Comment:
is it a memory leak if the timer isn't observed (because of an error), or is it ok?

Reply:
I don't think it is - I think this timer just gets garbage collected without adding any data to the metric.

@simonpasquier  can you  confirm there is no memory leak if we create/start a timer but never observe it?

Reply:
yep it should be fine.